src/settings/LocationLocations/LocationForm/RemoteStorageField.js

```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { useField } from 'react-final-form';
+
+import { Control, useRemoteStorageApi } from '../RemoteStorage';
+
+export const RemoteStorageField = ({ initialValues, checkLocationHasHoldingsOrItems }) => {
+  const { remoteMap, mappings, t } = useRemoteStorageApi();
+
+  const [isReadOnly, setIsReadOnly] = React.useState(true);
+
+  const locationId = initialValues?.id;
+
+  const field = useField('remoteId', { initialValue: remoteMap[locationId] });
+
+  React.useEffect(
+    () => {
+      // For new locations - initial state is NOT read-only
+      if (locationId === undefined) {
+        setIsReadOnly(false);
+        return;
+      }
+
+      // Initial state is read-only, until we get assured there are no Holdings or Items associated
```

Comment:
I suggest removing this comment according to the suggestion above. The code clearly expresses the intent that after successful API call to set read-only.

Reply:
done